pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/AbstractNode.java

```diff
@@ -236,40 +221,29 @@ public Node getNthParent(int n) {
         return (T) parentNode;
     }
 
-    /**
-     * Traverses up the tree to find all of the parent instances of type
-     * parentType
-     *
-     * @param parentType
-     *            classes which you want to find.
-     * @return List of parentType instances found.
-     */
+
     @Override
     public <T> List<T> getParentsOfType(Class<T> parentType) {
         List<T> parents = new ArrayList<>();
         Node parentNode = jjtGetParent();
         while (parentNode != null) {
-            if (parentNode.getClass() == parentType) {
+            if (parentType.isInstance(parentNode.getClass())) {
```

Comment:
This is the cause for the unit test failures (wouldn't have seen it otherwise):
It must be `parentType.isInstance(parentNode)` (without the getClass()) - now all the parent nodes have been matched (since every class is a class...).